### ISSUE TITLE
A declaration file cannot be imported without 'import type'

### ISSUE THREAD

**Issue by GreenmeisterDavid** [nuxt-hub/core#293]
**Describe the bug**
Adding the NuxtHub module to the latest version of Nuxt (3.13.2) gives an error when combined with vue-tsc typechecking. Output error;
```
 ERROR        
 ERROR(vue-tsc)  A declaration file cannot be imported without 'import type'. Did you mean to import an implementation file '/home/projects/github-tlv5io/node_modules/@nuxthub/core/dist/runtime/base/server/utils/hooks' instead?
 FILE  undefined:

undefined

[vue-tsc] Found 1 error. Watching for file changes.
```

**Steps to reproduce**
Reproduction on StackBlitz: https://stackblitz.com/edit/github-tlv5io

1. Start new nuxt project
2. Add the nuxtcore module using `npx nuxi module add hub `
3. Add typescript and vue-tsc dev deps
4. Set typechecking in nuxt.config.ts; 
```
typescript: {
    typeCheck: true,
  },
```
5. Run dev (`npm run dev`)

**Expected behavior**
No typescript errors.

**Comment by atinux** [comment]
This seems related directly to the module builder as Nitro generate a import script in `.nuxt/types/nitro-imports.d.ts`

![CleanShot 2024-09-24 at 17 38 25](https://github.com/user-attachments/assets/ec12de38-d97f-408a-8141-9022c21fb409)

**Comment by zacwebb** [comment]
Same here on `nuxt@3.13.2`.

Downgrading to `3.13.1` resolves it for now

**Comment by cernymatej** [comment]
Hi there, are there any workarounds available, or are you still on `3.13.1`?

**Comment by RihanArfan** [comment]
I can't replicate this on Nuxt v3.15.2 and @nuxthub/core 0.8.13. Are you still encountering this issue?

```
------------------------------
- Operating System: Linux
- Node Version:     v22.13.0
- Nuxt Version:     3.15.2
- CLI Version:      3.20.0
- Nitro Version:    2.10.4
- Package Manager:  pnpm@9.15.4
- Builder:          -
- User Config:      compatibilityDate, devtools, future, typescript, modules
- Runtime Modules:  @nuxthub/core@0.8.13
- Build Modules:    -
------------------------------
```

![Image](https://github.com/user-attachments/assets/d8b7e925-271b-4639-b356-14c2f50dec92)

**Comment by GreenmeisterDavid** [comment]
@RihanArfan Nope, it's been solved for me as well on these latest versions! Thanks 💚
